### PR TITLE
Auto-hide block palette addon

### PR DESCRIPTION
**Pull request by TheColaber**
**Resolves**

Requested in #1156

**Changes**

I wrote the addon.

**Tests**

Tested and works

### CODE REVIEW

File: addons/hide-flyout/addon.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    }
+  ],
+  "userstyles": [
+    {
+      "url": "style.css",
+      "matches": ["https://scratch.mit.edu/projects/*"]
+    }
+  ],
+
+  "tags": ["editor"],
```
Comment (WorldLanguages) [inline addons/hide-flyout/addon.json:23]: Recommended

File: addons/hide-flyout/userscript.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+export default async function ({ addon, global, console }) {
+  while (true) {
+    let flyOut = await addon.tab.waitForElement(".blocklyFlyout", { markAsSeen: true });
+    let state = true;
+    function mouseMove(e) {
+      let sidePos = document.querySelector(".blocklyToolboxDiv").getBoundingClientRect().x;
+      let scrollBar = document.querySelector(".blocklyFlyoutScrollbar");
+      if (e.clientX - sidePos < 310) {
```
Comment (WorldLanguages) [inline addons/hide-flyout/userscript.js:8]: What's 310? This feels like it will change screen to screen.

Comment (TheColaber) [reply addons/hide-flyout/userscript.js:8]: I think that it should be the same everywhere... This is the width of the flyout.

**Comment by TheColaber** [comment]
Ready to merge

**Comment by TheColaber** [comment]
Having a problem currently... Deleting blocks doesn't work! It used to... interesting...

**Review by WorldLanguages — APPROVED** [review]

**Comment by CuriousG2** [comment]
I tested this, but it seems to have a lot of bugs:

1. The animation seems to go over the stage when the stage is on the left, and you are moving a block.

2. When the stage is on the left in small mode the lock displays in the wrong place.

3. You can see the lock on the project page.

4. Going to the project page and back into the project breaks it.

5. When you go outside the project then go back in you can see the lock under the sounds and costumes tabs.

6. For some reason it creates a scroll bar on the right side of the screen.

This happens on Firefox, Chrome, and Edge.

**Comment by TheColaber** [comment]
Thanks for this @CuriousG2 I'll fix all these asap.

**Pull request merged**